odoo/addons/base/tests/test_sanitize.py

```diff
@@ -0,0 +1,13 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.tests.common import BaseCase
+from odoo.tools import html_sanitize
+
+
+class TestSanitize(BaseCase):
+    def test_comment_malformed(self):
+        html = '''<!-- malformed-close --!> <img src='x' onerror='alert(1)'></img> --> comment <!-- normal comment --> --> out of context balise --!>'''
+        html_result = html_sanitize(html)
+        self.assertFalse('alert(1)' in html_result)
+        self.assertTrue('--!>' in html_result)
```

Comment:
missing \n at the end of the file (you can configure your ide to automatically add it)